pandas/tseries/tests/test_tslib.py

```diff
@@ -949,6 +949,17 @@ def compare_local_to_utc(tz_didx, utc_didx):
                                   tslib.maybe_get_tz('Asia/Tokyo'))
         self.assert_numpy_array_equal(result, np.array([tslib.iNaT], dtype=np.int64))
 
+    def test_shift_months(self):
+        s = DatetimeIndex([Timestamp('2000-01-05 00:15:00'), Timestamp('2000-01-31 00:23:00'),
+                           Timestamp('2000-01-01'), Timestamp('2000-02-29'), Timestamp('2000-12-31')])
+        for years in [-1, 0, 1]:
```

Comment:
didn't add here because the `tslib` method just operates on the ints, but I did add names to some of the existing `DateOffset` tests